### ISSUE TITLE
fasta: Issues with calls through static getters 

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#28562]
A number of co19 tests that test calls through static getters like these:

```dart
static get m => () { };

m();
```

are compiled incorrectly by dartk: either `NoSuchMethod` is emitted or invocation is compiled as an instance call:

```
co19/Language/Expressions/Function_Invocation/Function_Expression_Invocation/not_a_function_expression_t01
co19/Language/Expressions/Function_Invocation/Unqualified_Invocation/function_expr_invocation_t03
co19/Language/Expressions/Function_Invocation/Unqualified_Invocation/function_expr_invocation_t04
co19/Language/Expressions/Method_Invocation/Ordinary_Invocation/accessible_instance_member_t05
co19/Language/Expressions/Method_Invocation/Ordinary_Invocation/evaluation_t07
co19/Language/Expressions/Method_Invocation/Ordinary_Invocation/method_lookup_failed_t18
```

/cc #28263

**Comment by mraleph** [comment]
It looks like some of these tests are passing, but in reality the code generated by fasta is incorrect.

```dart
class C {
  static get staticget => () {return 3;};
}

main() {
  C.staticget();
}
```

compiles down to 

```
  class C extends core::Object {
    constructor •() → void
      : super core::Object::•()
      ;
    static get staticget() → dynamic
      return () → dynamic {
        return 3;
      };
  }
  static method main() → dynamic {
    core::print(acc::C::staticget());
  }
```

**Comment by peter-ahe-google** [comment]
I have fixed this locally.

**Comment by peter-ahe-google** [comment]
Fixed by cbcb7d4885b3a380c05140dee0fcaf8840a13558.
